src/main/scala/com/mozilla/telemetry/streaming/FederatedLearningSearchOptimizer.scala

```diff
@@ -40,31 +40,47 @@ object FederatedLearningSearchOptimizer extends StreamingJobBase {
       .select("value")
 
     val query = optimize(pings,
-      opts.checkpointPath(), opts.modelOutputBucket(), opts.modelOutputKey(), opts.stateCheckpointPath(),
-      opts.stateBootstrapFilePath.get, Clock.systemUTC(), opts.windowOffsetMinutes(), opts.raiseOnError(),
-      opts.s3EndpointOverride.get)
+      opts.checkpointPath(), opts.modelOutputBucket(),
+      opts.modelOutputKey(), opts.modelBranch(),
+      opts.stateCheckpointPath(), opts.stateBootstrapFilePath.get,
+      Clock.systemUTC(), opts.windowOffsetMinutes(),
+      opts.raiseOnError(), opts.s3EndpointOverride.get)
 
     query.awaitTermination()
   }
 
   def optimize(pings: DataFrame, checkpointPath: String, modelOutputBucket: String, modelOutputKey: String, // scalastyle:ignore
-               stateCheckpointPath: String, stateBootstrapFilePath: Option[String] = None,
+               modelBranch: Int, stateCheckpointPath: String, stateBootstrapFilePath: Option[String] = None,
                clock: Clock, windowOffsetMin: Int, raiseOnError: Boolean = false, s3EndpointOverride: Option[String] = None): StreamingQuery = {
-    val aggregates = aggregate(pings, clock, windowOffsetMin, raiseOnError)
+    val aggregates = aggregate(pings, modelBranch, clock, windowOffsetMin, raiseOnError)
     writeUpdates(aggregates, checkpointPath, modelOutputBucket, modelOutputKey, stateCheckpointPath, stateBootstrapFilePath, s3EndpointOverride)
   }
 
-  def aggregate(pings: DataFrame, clock: Clock, windowOffsetMin: Int, raiseOnError: Boolean = false): Dataset[FrecencyUpdateAggregate] = {
+  def aggregate(pings: DataFrame, modelBranch: Int, clock: Clock, windowOffsetMin: Int, raiseOnError: Boolean = false): Dataset[FrecencyUpdateAggregate] = {
     import pings.sparkSession.implicits._
 
+
+    val variationTarget: String = modelBranch match {
+      case 1 => "model1"
+      case 2 => "model2"
+      case 3 => "model3"
+      case 4 => "model4"
+    }
```

Comment:
What's the value in using this int to String mapping? This mapping is done only once -- seems simpler to just pass in the actual string to the job

Reply:
There isn't one. :)

I've simplified this in fa3787d to use the full model string name instead of doing the conversion from int.